src/golang/lib/collections/workflow/sqlite.go

```diff
@@ -53,26 +53,39 @@ func (r *sqliteReaderImpl) GetWorkflowsWithLatestRunResult(
 	ctx context.Context,
 	organizationId string,
 	db database.Database,
-) ([]latestWorkflowResponse, error) {
+) ([]LatestWorkflowResponse, error) {
 	// Get workflow metadata (id, name, description, creation time, last run time, and last run status)
 	// for all workflows whose `organization_id` is `organizationId` ordered by when the workflow was created.
 	// Get the last run DAG by getting the max created_at timestamp for all workflow DAGs associated with each
 	// workflow in the organization.
-	query := `SELECT wf.id AS id, wf.name AS name, 
-		 wf.description AS description, wf.created_at AS created_at, 
-		 wfdr.created_at AS last_run_at, wfdr.status as status 
-		 FROM workflow AS wf, app_user, workflow_dag AS wfd, workflow_dag_result AS wfdr, 
-		 (SELECT wf.id AS id, MAX(wfdr.created_at) AS last_run_at 
-		  FROM workflow AS wf, app_user, workflow_dag AS wfd, workflow_dag_result AS wfdr 
-		  WHERE app_user.organization_id = $1 
-		  AND wf.user_id = app_user.id AND wfd.workflow_id = wf.id AND wfdr.workflow_dag_id = wfd.id 
-		  GROUP BY wf.id) AS wflr 
-		 WHERE app_user.organization_id = $1 
-		 AND wf.user_id = app_user.id AND wfd.workflow_id = wf.id AND wfdr.workflow_dag_id = wfd.id AND 
-		 wf.id = wflr.id AND wfdr.created_at = wflr.last_run_at 
-		 ORDER BY created_at DESC;`
+	query := `
+		WITH workflow_results AS
```

Comment:
There's a lot of rewriting her to make things simpler imo. It's easier to understand views than subqueries, when the subquery isn't necessary.